remotes/fixup.go

```diff
@@ -43,37 +44,64 @@ func FixupBundle(ctx context.Context, b *bundle.Bundle, ref reference.Named, res
 
 	// Fixup invocation images
 	if len(b.InvocationImages) != 1 {
-		return fmt.Errorf("only one invocation image supported for bundle %q", ref)
+		return nil, fmt.Errorf("only one invocation image supported for bundle %q", ref)
 	}
-	if b.InvocationImages[0].BaseImage, err = fixupImage(ctx, b.InvocationImages[0].BaseImage, cfg, events, cfg.invocationImagePlatformFilter); err != nil {
-		return err
+
+	relocationMap := relocation.ImageRelocationMap{}
+	if err := fixupImage(ctx, &b.InvocationImages[0].BaseImage, relocationMap, cfg, events, cfg.invocationImagePlatformFilter); err != nil {
+		return nil, err
 	}
 	// Fixup images
 	for name, original := range b.Images {
-		if original.BaseImage, err = fixupImage(ctx, original.BaseImage, cfg, events, cfg.componentImagePlatformFilter); err != nil {
-			return err
+		if err := fixupImage(ctx, &original.BaseImage, relocationMap, cfg, events, cfg.componentImagePlatformFilter); err != nil {
```

Comment:
I don't understand why the output from `fixupImage` is sent to a separate goroutine in order to be printed to stderr. Since the calls to `fixupImage` are executed serially, why can't they just print to stderr themselves?

Reply:
As the `fixup` will push or mount several images, we thought that having a more sophisticated event base mechanism will help the consumer of this library to create a good UX on this. For example with `docker/app` we use `https://github.com/morikuni/aec` to have a nice UI in the term. Does that make sense?

Reply:
I suppose so. The only issue I can think of immediately is if one of these event listeners blocks, it will hold up the termination of the whole process, but I guess that's ok. The user can always Ctrl-C.